contrib/win32/win32compat/misc.c

```diff
@@ -482,10 +482,10 @@ strmode(mode_t mode, char *p)
 	 * As of now we are keeping "*" for everything.
 	 * TODO - figure out if there is a better option
 	 */
-	const char *permissions = "********* ";
-	strncpy(p, permissions, strlen(permissions) + 1);
-
-	p = p + strlen(p);
+	const char *permissions = "********* ";	
+	for(int i = 0; i < strlen(permissions); i++)
```

Comment:
strncpy is giving a warning to use strncpy_s..
To use strncpy_s we need to know the size of p.. since p is a char*, the caller can pass pointer to a memory (or) an array so we can't decide the size here.. so to avoid the warning.. changed it to for loop